### PR TITLE
Clarify anonymous authentication support 

### PR DESCRIPTION
**Pull request by scrawfor99**
### Description
This change clarifies the anonymous authentication feature and its configuration. 

### Issues Resolved
Fixes https://github.com/opensearch-project/documentation-website/issues/627

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by scrawfor99** [comment]
It says that "[Anonymous authentication](../access-control/anonymous-authentication.md)" is a dead link inside of the configuration.md file. I don't thin that is correct. Perhaps the doc team can help me fix this.

File: _security/access-control/anonymous-authentication.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+http:
+  anonymous_auth_enabled: <true|false>
+  ...
+```
+
+The following table describes the `anonymous_auth_enabled` setting. For more information, see the [configuration]({{site.url}}{{site.baseurl}}/security/configuration/configuration/) file overview. 
+
+| Setting | Description |
+| :--- | :--- |
+| `anonymous_auth_enabled` | Either enables or disables anonymous authentication. When you enable anonymous authentication, all defined HTTP authenticators are non-challenging. See [The challenge setting]({{site.url}}{{site.baseurl}}/security/authentication-backends/basic-authc/#the-challenge-setting). |
```
Comment (kolchfa-aws) [inline _security/access-control/anonymous-authentication.md:28]: Where should the HTTP authenticators link to? Is there a doc page that describes them?

**Review by kolchfa-aws — APPROVED** [review]
Thanks, @scrawfor99! Some helpful hints: 
- specify the parent page in order for the child page to be visible
- when linking, we use an absolute path. In this path, the root directory is specified by `{{site.url}}{{site.baseurl}}`, the directory immediately under the root directory does not have an underscore, and the file name itself does not have an extension. This is how the link is formatted on the doc site
- the syntax highlighting specified for yaml files is `yaml`
- variables and settings are surrounded with backticks instead of quotes.
Hope this helps. Thanks for providing a lot of great documentation!

Also, let me know what version this is applicable to or add a version label so we can backport to the correct versions. Thanks!

File: _security/access-control/anonymous-authentication.md (hunk 1 above)
Comment (scrawfor99) [reply _security/access-control/anonymous-authentication.md:28]: The Http Authenticators are described in the general authentication backend page.

Comment (kolchfa-aws) [reply _security/access-control/anonymous-authentication.md:28]: Thanks; we're already linking to that page so it should be good then.

**Comment by scrawfor99** [comment]
@kolchfa-aws @hdhalter Should be all set :)

File: _security/access-control/anonymous-authentication.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+---
+layout: default
+title: Anonymous authentication
+parent: Access control
+nav_order: 145
+---
+
+# Anonymous authentication
+
+The Security plugin supports anonymous authentication where a user is able to access a cluster without providing credentials. This is useful in cases where you want lots of people to be able to access your cluster with a common set of privileges.
```
Comment (natebower) [inline _security/access-control/anonymous-authentication.md:10]:
```suggestion
The Security plugin supports anonymous authentication, through which a user is able to access a cluster without providing credentials. This is useful in cases where you want lots of people to be able to access your cluster with a common set of privileges. 
```

File: _security/access-control/anonymous-authentication.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+# Anonymous authentication
+
+The Security plugin supports anonymous authentication where a user is able to access a cluster without providing credentials. This is useful in cases where you want lots of people to be able to access your cluster with a common set of privileges. 
+
+## Configuration
+
+To enable anonymous authentication, you need to modify the `config.yml` file inside the `opensearch-security` configuration subdirectory of your cluster.
+
+In the `config.yml` file, there is an `http` section which includes the `anonymous_auth_enabled` setting:
```
Comment (natebower) [inline _security/access-control/anonymous-authentication.md:16]:
```suggestion
In the `config.yml` file, there is an `http` section, which includes the `anonymous_auth_enabled` setting: 
```

File: _security/access-control/anonymous-authentication.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  reserved: false
+  hidden: false
+  backend_roles: ["opendistro_security_anonymous_backendrole"]
+  hosts: []
+```
+{% include copy.html %}
+
+Notice that the role is mapped to `opendistro_security_anonymous_backendrole`, which means that all users with the anonymous user backend role will have these privileges. 
+
+Alternatively, you can complete these steps using the REST API or OpenSearch Security Dashboards.
```
Comment (natebower) [inline _security/access-control/anonymous-authentication.md:73]: Is "Security" supposed to be there?

File: _security/configuration/configuration.md [hunk 5]
```diff
@@ -46,15 +46,14 @@ http:
...
+| `anonymous_auth_enabled` | Either enables or disables anonymous authentication. When `true`, HTTP authenticators try to find user credentials in the HTTP request. If credentials are found, the user is authenticated. If none are found, the user is authenticated as an "anonymous" user. This user then has the username `anonymous` and one role named `anonymous_backendrole`. When you enable anonymous authentication, all defined HTTP authenticators are non-challenging. For more information, see [The challenge setting]({{site.url}}{{site.baseurl}}/security/authentication-backends/basic-authc/#the-challenge-setting). |
```
Comment (natebower) [inline _security/configuration/configuration.md:51]:
```suggestion
| `anonymous_auth_enabled` | Either enables or disables anonymous authentication. When `true`, HTTP authenticators try to find user credentials in the HTTP request. If credentials are found, the user is authenticated. If none are found, the user is authenticated as an _anonymous_ user. This user then has the username `anonymous` and one role named `anonymous_backendrole`. When you enable anonymous authentication, all defined HTTP authenticators are non-challenging. For more information, see [The challenge setting]({{site.url}}{{site.baseurl}}/security/authentication-backends/basic-authc/#the-challenge-setting). |
```

File: _security/configuration/configuration.md [hunk 6]
```diff
@@ -46,15 +46,14 @@ http:
...
+| `xff` | Used to configure proxy-based authentication. For more information about this backend, see [Proxy-based authentication]({{site.url}}{{site.baseurl}}/security/authentication-backends/proxy/).                                                                                                                                                                                                                                                                                                                                                                                                                           |
 
-If you disable anonymous authentication, the Security plugin won't initialize if you have not provided at least one `authc`.
+For instructions to configure anonymous authentication, see [Anonymous authentication]({{site.url}}{{site.baseurl}}/security/access-control/anonymous-authentication/).
```
Comment (natebower) [inline _security/configuration/configuration.md:54]:
```suggestion
For instructions on how to configure anonymous authentication, see [Anonymous authentication]({{site.url}}{{site.baseurl}}/security/access-control/anonymous-authentication/).  
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _security/access-control/anonymous-authentication.md (hunk 4 above)
Comment (kolchfa-aws) [reply _security/access-control/anonymous-authentication.md:73]: Not necessary. Removing.

Comment (kolchfa-aws) [inline _security/access-control/anonymous-authentication.md:73]:
```suggestion
Alternatively, you can complete these steps using the REST API or OpenSearch Dashboards. 
```

**Pull request merged**